src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1544,6 +1548,170 @@ GetSequencesFromAttrDef(Oid attrdefOid)
 }
 
 
+/*
+ * GetDependentFunctionsWithRelation returns the dependent functions for the
+ * given relation id.
+ */
+List *
+GetDependentFunctionsWithRelation(Oid relationId)
+{
+	List *attrdefResult = NIL;
+	List *constraintResult = NIL;
+	List *dependentFunctionOids = NIL;
+	ScanKeyData key[2];
+	HeapTuple tup;
+
+	Relation depRel = table_open(DependRelationId, AccessShareLock);
+
+	ScanKeyInit(&key[0],
+				Anum_pg_depend_refclassid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(RelationRelationId));
+	ScanKeyInit(&key[1],
+				Anum_pg_depend_refobjid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(relationId));
+
+	SysScanDesc scan = systable_beginscan(depRel, DependReferenceIndexId, true,
+										  NULL, 2, key);
+
+	while (HeapTupleIsValid(tup = systable_getnext(scan)))
+	{
+		Form_pg_depend deprec = (Form_pg_depend) GETSTRUCT(tup);
+
+		/*
+		 * Add columns with default value
+		 */
+		if (deprec->classid == AttrDefaultRelationId &&
+			deprec->objsubid == 0 &&
+			deprec->refobjsubid != 0 &&
```

Comment:
what is the meaning of `deprec->objsubid == 0 && deprec->refobjsubid != 0` here?

Reply:
objsubid is nonzero only for table columns and zero for anything else. Since we are trying to find a dependency from the column of a table to function we've added such a check. We may `omit deprec->objsubid == 0` part , but I'm more inclined to keep it to not to get any unintentional dependency.

Reply:
can you maybe add this to code? Because this is something we'd forget :)


bool IsColumnDependency =  deprec->refobjsubid != 0